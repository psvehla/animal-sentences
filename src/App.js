import React, { Component } from 'react';
import axios from 'axios';
import NumericInput from 'react-numeric-input';
import './App.css';

class App extends Component {

  constructor(props) {
    super(props);
    this.state = {
      selectedFile: null,
      prefixLen: 5,
      suffixLen: 1,
      numberOfSentences: 10,
      output: "Please upload some plain text.",
    }

    this.onChangeHandler = this.onChangeHandler.bind(this);
    this.onPrefixChangeHandler = this.onPrefixChangeHandler.bind(this);
  }

  onChangeHandler=event=>{
    this.setState({
      selectedFile: event.target.files[0],
      loaded: 0,
    })
  }

  onClickHandler=()=>{
    const data = new FormData();
    data.append('file', this.state.selectedFile);
    data.append('prefixLen', this.state.prefixLen);
    data.append('suffixLen', this.state.suffixLen);
    data.append('numberOfSentences', this.state.numberOfSentences);

    axios.post("http://localhost:8080/api/generate", data, {}).then(res => {
      this.setState({
        output: res.data,
      });
    })
  }

  onPrefixChangeHandler=event=>{
    console.log("onPrefixChangeHandler called" + event);
    this.setState({
      prefixLen: event,
    })
  }

  onSuffixChangeHandler=event=>{
    console.log("onSuffixChangeHandler called" + event);
    this.setState({
      suffixLen: event,
    })
  }

  onNumberOfSentencesChangeHandler=event=>{
    console.log("onSuffixChangeHandler called" + event);
    this.setState({
      numberOfSentences: event,
    })
  }

  render() {
    return (
      <div className="App">
        <header className="App-header">
          <div id="output">{this.state.output}</div>
          <div className="container">
    	      <div className="row">
          	  <div className="offset-md-3 col-md-6">
                <label>Prefix Length</label>
                <NumericInput id="prefixLen" type="number" onChange={this.onPrefixChangeHandler} value={this.state.prefixLen} min={1} max={100} />
                <label>Suffix Length</label>
                <NumericInput id="suffixLen" type="number" onChange={this.onSuffixChangeHandler} value={this.state.suffixLen} min={1} max={10} />
                <label>Number of Sentences</label>
                <NumericInput id="numberOfSentences" type="number" onChange={this.onNumberOfSentencesChangeHandler} value={this.state.numberOfSentences} min={1} max={1000} />
                <div className="form-group files">
                  <label>Upload Your File </label>
                  <input type="file" className="form-control" multiple={false} onChange={this.onChangeHandler}/>
                </div>
                <button type="button" className="btn btn-success btn-block" onClick={this.onClickHandler}>Upload</button>
    	        </div>
            </div>
          </div>
          <a
            className="App-link"
            href="https://reactjs.org"
            target="_blank"
            rel="noopener noreferrer"
          >
            Learn React
          </a>
        </header>
      </div>
    );
  }
}

export default App;
