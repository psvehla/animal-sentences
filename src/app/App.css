.App {
  text-align: center;
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
  padding: 35px;
}

.App-link {
  color: #61dafb;
}

h1 {
  padding: 35px;
}

label {
  padding: 0px 35px 0px 35px;
}

#output {
  padding: 35px;
  color: yellow;
}
