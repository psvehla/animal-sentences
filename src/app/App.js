import React, { Component } from 'react';
import axios from 'axios';
import NumericInput from 'react-numeric-input';
import FileUpload from '../file-upload/FileUpload';
import './App.css';

class App extends Component {

  constructor(props) {
    super(props);
    this.state = {
      selectedFile: null,
      prefixLen: 5,
      suffixLen: 1,
      numberOfSentences: 10,
      output: "Please upload some plain text.",
    }
  }

  checkMimeType=(event)=>{

    let file = event.target.files[0];

    if (file && file.type !== 'text/plain') {
      this.setState({
        output: "Only plain text files are supported.",
      })

      return false;
    }

    return true;
  }

  checkFileSize=(event)=>{

    let file = event.target.files[0];

    if (file && file.size > 1048576 ) {
      this.setState({
        output: "Your file is too big, please try another one.",
      })

      return false;
    }

    return true;
  }

  onChangeHandler=event=>{
    if (this.checkMimeType(event) && this.checkFileSize(event)) {
      this.setState({
        selectedFile: event.target.files[0],
        loaded: 0,
      })
    }
  }

  onClickHandler=()=>{
    const data = new FormData();
    data.append('file', this.state.selectedFile);
    data.append('prefixLen', this.state.prefixLen);
    data.append('suffixLen', this.state.suffixLen);
    data.append('numberOfSentences', this.state.numberOfSentences);

    axios.post(process.env.REACT_APP_SENTENCE_SERVICE_ENDPOINT, data, {})
      .then(res => {
        this.setState({
          output: res.data,
        });
      })
      .catch((error) => {
        if (error.response) {
          this.setState({
            output: "Your file is invalid. Please upload a plain text one.",
          })
          console.error(error.response.data);
          console.error(error.response.status);
          console.error(error.response.headers);
        }
        else if (error.request) {
          console.error(error.request);
        }
        else {
          console.error('Error', error.message);
        }
        console.error(error.config);
      });
  }

  onPrefixChangeHandler=event=>{
    this.setState({
      prefixLen: event,
    })
  }

  onSuffixChangeHandler=event=>{
    this.setState({
      suffixLen: event,
    })
  }

  onNumberOfSentencesChangeHandler=event=>{
    this.setState({
      numberOfSentences: event,
    })
  }

  render() {
    return (
      <div className="App">
        <header className="App-header">
          <h1>Sentence Generator</h1>
          <p>
            This application generates sentences based on the provided text. It uses a Markov chain algorithm, you can adjust the prefix and suffix size of the algorithm, though reasonable
            defaults have been chosen for you. Feel free to experiment!
          </p>
          <div id="output">{this.state.output}</div>
          <div className="container">
    	      <div className="row">
          	  <div className="offset-md-3 col-md-6">
                <label>Prefix Length</label>

                <NumericInput id="prefixLen" style={{
                  input: {
                    width: 200
                  }
                }} type="number" onChange={this.onPrefixChangeHandler} value={this.state.prefixLen} min={1} max={100} />

              </div>
            </div>
            <div className="row">
          	  <div className="offset-md-3 col-md-6">
                <label>Suffix Length</label>

                <NumericInput id="suffixLen" style={{
                  input: {
                    width: 200,
                  }
                }} type="number" onChange={this.onSuffixChangeHandler} value={this.state.suffixLen} min={1} max={10} />

              </div>
            </div>
            <div className="row">
          	  <div className="offset-md-3 col-md-6">
                <label>Number of Sentences</label>
              </div>
            </div>
            <div className="row" style={{marginBottom: '35px'}}>
          	  <div className="offset-md-3 col-md-6">
                <NumericInput id="numberOfSentences" style={{
                  input: {
                    width: 200,
                  }
                }} type="number" onChange={this.onNumberOfSentencesChangeHandler} value={this.state.numberOfSentences} min={1} max={1000} />
              </div>
            </div>
            <div className="row">
              <div className="offset-md-3 col-md-6">
                <FileUpload onChange={this.onChangeHandler} onClick={this.onClickHandler} />
    	        </div>
            </div>
          </div>
        </header>
      </div>
    );
  }
}

export default App;
