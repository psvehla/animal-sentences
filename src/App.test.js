import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import NumericInput from 'react-numeric-input';
import App from './App';

it('renders correctly', () => {

  const tree = renderer
    .create(<App page="https://localhost:3000">AnimalSentences</App>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

test('test prefixLen input', () => {

  const TEST_VALUE = 10
  const changePrefixLenMock = jest.fn();

  const event = {
    preventDefault() {},
    target: { value: TEST_VALUE}
  };

  const component = shallow(<NumericInput onChange={changePrefixLenMock} />);
  component.find('input').simulate('change', event);
  expect(changePrefixLenMock).toHaveBeenCalledTimes(1);
});

test('test suffixLen input', () => {

  const TEST_VALUE = 10
  const changeSuffixLenMock = jest.fn();

  const event = {
    preventDefault() {},
    target: { value: TEST_VALUE}
  };

  const component = shallow(<NumericInput onChange={changeSuffixLenMock} />);
  component.find('input').simulate('change', event);
  expect(changeSuffixLenMock).toHaveBeenCalledTimes(1);
});

test('test numberOfSentences input', () => {

  const TEST_VALUE = 10
  const changeNumberOfSentencesMock = jest.fn();

  const event = {
    preventDefault() {},
    target: { value: TEST_VALUE}
  };

  const component = shallow(<NumericInput onChange={changeNumberOfSentencesMock} />);
  component.find('input').simulate('change', event);
  expect(changeNumberOfSentencesMock).toHaveBeenCalledTimes(1);
});
