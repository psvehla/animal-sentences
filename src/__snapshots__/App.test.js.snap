// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <h1>
      Sentence Generator
    </h1>
    <p>
      This application generates sentences based on the provided text. It uses a Markov chain algorithm, you can adjust the prefix and suffix size of the algorithm, though reasonable defaults have been chosen for you. Feel free to experiment!
    </p>
    <div
      id="output"
    >
      Please upload some plain text.
    </div>
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="offset-md-3 col-md-6"
        >
          <label>
            Prefix Length
          </label>
          <span
            className="react-numeric-input"
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            style={
              Object {
                "display": "inline-block",
                "position": "relative",
              }
            }
          >
            <input
              id="prefixLen"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onInput={[Function]}
              onKeyDown={[Function]}
              onSelect={[Function]}
              style={
                Object {
                  "WebkitAppearance": "none",
                  "border": "1px solid #ccc",
                  "borderRadius": 2,
                  "boxSizing": "border-box",
                  "display": "block",
                  "fontSize": "inherit",
                  "lineHeight": "normal",
                  "paddingLeft": 4,
                  "paddingRight": "3ex",
                  "width": 200,
                }
              }
              type="text"
              value="5"
            />
            <b
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchStart={[Function]}
              style={
                Object {
                  "background": "rgba(0,0,0,.1)",
                  "borderColor": "rgba(0,0,0,.1)",
                  "borderRadius": "2px 2px 0 0",
                  "borderStyle": "solid",
                  "borderWidth": "1px 1px 0 1px",
                  "bottom": "50%",
                  "boxShadow": "-1px -1px 3px rgba(0,0,0,.1) inset,1px 1px 3px rgba(255,255,255,.7) inset",
                  "cursor": "default",
                  "position": "absolute",
                  "right": 2,
                  "textAlign": "center",
                  "top": 2,
                  "transition": "all 0.1s",
                  "width": "2.26ex",
                }
              }
            >
              <i
                style={
                  Object {
                    "borderColor": "transparent transparent rgba(0, 0, 0, 0.7)",
                    "borderStyle": "solid",
                    "borderWidth": "0 0.6ex 0.6ex 0.6ex",
                    "height": 0,
                    "left": "50%",
                    "margin": "-0.3ex 0 0 -0.56ex",
                    "position": "absolute",
                    "top": "50%",
                    "width": 0,
                  }
                }
              />
            </b>
            <b
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchStart={[Function]}
              style={
                Object {
                  "background": "rgba(0,0,0,.1)",
                  "borderColor": "rgba(0,0,0,.1)",
                  "borderRadius": "0 0 2px 2px",
                  "borderStyle": "solid",
                  "borderWidth": "0 1px 1px 1px",
                  "bottom": 2,
                  "boxShadow": "-1px -1px 3px rgba(0,0,0,.1) inset,1px 1px 3px rgba(255,255,255,.7) inset",
                  "cursor": "default",
                  "position": "absolute",
                  "right": 2,
                  "textAlign": "center",
                  "top": "50%",
                  "transition": "all 0.1s",
                  "width": "2.26ex",
                }
              }
            >
              <i
                style={
                  Object {
                    "borderColor": "rgba(0, 0, 0, 0.7) transparent transparent",
                    "borderStyle": "solid",
                    "borderWidth": "0.6ex 0.6ex 0 0.6ex",
                    "height": 0,
                    "left": "50%",
                    "margin": "-0.3ex 0 0 -0.56ex",
                    "position": "absolute",
                    "top": "50%",
                    "width": 0,
                  }
                }
              />
            </b>
          </span>
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="offset-md-3 col-md-6"
        >
          <label>
            Suffix Length
          </label>
          <span
            className="react-numeric-input"
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            style={
              Object {
                "display": "inline-block",
                "position": "relative",
              }
            }
          >
            <input
              id="suffixLen"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onInput={[Function]}
              onKeyDown={[Function]}
              onSelect={[Function]}
              style={
                Object {
                  "WebkitAppearance": "none",
                  "border": "1px solid #ccc",
                  "borderRadius": 2,
                  "boxSizing": "border-box",
                  "display": "block",
                  "fontSize": "inherit",
                  "lineHeight": "normal",
                  "paddingLeft": 4,
                  "paddingRight": "3ex",
                  "width": 200,
                }
              }
              type="text"
              value="1"
            />
            <b
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchStart={[Function]}
              style={
                Object {
                  "background": "rgba(0,0,0,.1)",
                  "borderColor": "rgba(0,0,0,.1)",
                  "borderRadius": "2px 2px 0 0",
                  "borderStyle": "solid",
                  "borderWidth": "1px 1px 0 1px",
                  "bottom": "50%",
                  "boxShadow": "-1px -1px 3px rgba(0,0,0,.1) inset,1px 1px 3px rgba(255,255,255,.7) inset",
                  "cursor": "default",
                  "position": "absolute",
                  "right": 2,
                  "textAlign": "center",
                  "top": 2,
                  "transition": "all 0.1s",
                  "width": "2.26ex",
                }
              }
            >
              <i
                style={
                  Object {
                    "borderColor": "transparent transparent rgba(0, 0, 0, 0.7)",
                    "borderStyle": "solid",
                    "borderWidth": "0 0.6ex 0.6ex 0.6ex",
                    "height": 0,
                    "left": "50%",
                    "margin": "-0.3ex 0 0 -0.56ex",
                    "position": "absolute",
                    "top": "50%",
                    "width": 0,
                  }
                }
              />
            </b>
            <b
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchStart={[Function]}
              style={
                Object {
                  "background": "rgba(0,0,0,.1)",
                  "borderColor": "rgba(0,0,0,.1)",
                  "borderRadius": "0 0 2px 2px",
                  "borderStyle": "solid",
                  "borderWidth": "0 1px 1px 1px",
                  "bottom": 2,
                  "boxShadow": "-1px -1px 3px rgba(0,0,0,.1) inset,1px 1px 3px rgba(255,255,255,.7) inset",
                  "cursor": "default",
                  "position": "absolute",
                  "right": 2,
                  "textAlign": "center",
                  "top": "50%",
                  "transition": "all 0.1s",
                  "width": "2.26ex",
                }
              }
            >
              <i
                style={
                  Object {
                    "borderColor": "rgba(0, 0, 0, 0.7) transparent transparent",
                    "borderStyle": "solid",
                    "borderWidth": "0.6ex 0.6ex 0 0.6ex",
                    "height": 0,
                    "left": "50%",
                    "margin": "-0.3ex 0 0 -0.56ex",
                    "position": "absolute",
                    "top": "50%",
                    "width": 0,
                  }
                }
              />
            </b>
          </span>
        </div>
      </div>
      <div
        className="row"
        style={
          Object {
            "marginBottom": "35px",
          }
        }
      >
        <div
          className="offset-md-3 col-md-6"
        >
          <label>
            Number of Sentences
          </label>
          <span
            className="react-numeric-input"
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            style={
              Object {
                "display": "inline-block",
                "position": "relative",
              }
            }
          >
            <input
              id="numberOfSentences"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onInput={[Function]}
              onKeyDown={[Function]}
              onSelect={[Function]}
              style={
                Object {
                  "WebkitAppearance": "none",
                  "border": "1px solid #ccc",
                  "borderRadius": 2,
                  "boxSizing": "border-box",
                  "display": "block",
                  "fontSize": "inherit",
                  "lineHeight": "normal",
                  "paddingLeft": 4,
                  "paddingRight": "3ex",
                  "width": 200,
                }
              }
              type="text"
              value="10"
            />
            <b
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchStart={[Function]}
              style={
                Object {
                  "background": "rgba(0,0,0,.1)",
                  "borderColor": "rgba(0,0,0,.1)",
                  "borderRadius": "2px 2px 0 0",
                  "borderStyle": "solid",
                  "borderWidth": "1px 1px 0 1px",
                  "bottom": "50%",
                  "boxShadow": "-1px -1px 3px rgba(0,0,0,.1) inset,1px 1px 3px rgba(255,255,255,.7) inset",
                  "cursor": "default",
                  "position": "absolute",
                  "right": 2,
                  "textAlign": "center",
                  "top": 2,
                  "transition": "all 0.1s",
                  "width": "2.26ex",
                }
              }
            >
              <i
                style={
                  Object {
                    "borderColor": "transparent transparent rgba(0, 0, 0, 0.7)",
                    "borderStyle": "solid",
                    "borderWidth": "0 0.6ex 0.6ex 0.6ex",
                    "height": 0,
                    "left": "50%",
                    "margin": "-0.3ex 0 0 -0.56ex",
                    "position": "absolute",
                    "top": "50%",
                    "width": 0,
                  }
                }
              />
            </b>
            <b
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchStart={[Function]}
              style={
                Object {
                  "background": "rgba(0,0,0,.1)",
                  "borderColor": "rgba(0,0,0,.1)",
                  "borderRadius": "0 0 2px 2px",
                  "borderStyle": "solid",
                  "borderWidth": "0 1px 1px 1px",
                  "bottom": 2,
                  "boxShadow": "-1px -1px 3px rgba(0,0,0,.1) inset,1px 1px 3px rgba(255,255,255,.7) inset",
                  "cursor": "default",
                  "position": "absolute",
                  "right": 2,
                  "textAlign": "center",
                  "top": "50%",
                  "transition": "all 0.1s",
                  "width": "2.26ex",
                }
              }
            >
              <i
                style={
                  Object {
                    "borderColor": "rgba(0, 0, 0, 0.7) transparent transparent",
                    "borderStyle": "solid",
                    "borderWidth": "0.6ex 0.6ex 0 0.6ex",
                    "height": 0,
                    "left": "50%",
                    "margin": "-0.3ex 0 0 -0.56ex",
                    "position": "absolute",
                    "top": "50%",
                    "width": 0,
                  }
                }
              />
            </b>
          </span>
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="offset-md-3 col-md-6"
        >
          <div
            className="form-group files"
          >
            <label
              id="upload-label"
            >
              Upload Your File 
            </label>
            <input
              className="form-control"
              multiple={false}
              onChange={[Function]}
              type="file"
            />
          </div>
          <button
            className="btn btn-success btn-block"
            onClick={[Function]}
            type="button"
          >
            Upload
          </button>
        </div>
      </div>
    </div>
  </header>
</div>
`;
